src/callbacks_step/glm_speed.jl

```diff
@@ -6,26 +6,30 @@
 #! format: noindent
 
 """
-    GlmSpeedCallback(; glm_scale=0.5, cfl)
+    GlmSpeedCallback(; glm_scale=0.5, cfl, semi_indices=())
 
 Update the divergence cleaning wave speed `c_h` according to the time step
 computed in [`StepsizeCallback`](@ref) for the ideal GLM-MHD equations.
 The `cfl` number should be set to the same value as for the time step size calculation. The
 `glm_scale` ensures that the GLM wave speed is lower than the fastest physical waves in the MHD
 solution and should thus be set to a value within the interval [0,1]. Note that `glm_scale = 0`
 deactivates the divergence cleaning.
+In case of coupled semidiscretizations, specify for which semi_index, i.e. index of the
+semidiscretization, the divergence cleaning should be applied.
 """
 struct GlmSpeedCallback{RealT <: Real}
     glm_scale::RealT
     cfl::RealT
+    semi_indices::Tuple
```

Comment:
`Tuple` is not a concrete type, making this struct perpetually type unstable.

I think that storing the values as a tuple is not required for performance, so if you just change this to
```suggestion
    semi_indices::Vector{Int}
```
I think you're ok. In that case, you can just call `collect(semi_indices)` in the constructor, which will convert any one-dimensional collection (tuple, svector, vector etc.) into a `Vector`.